server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -220,6 +220,13 @@ final class DefaultSearchContext extends SearchContext {
             );
             queryBoost = request.indexBoost();
             this.lowLevelCancellation = lowLevelCancellation;
+            SearchSourceBuilder builder = request.source();
```

Comment:
Not sure if this is a reasonable expectation that source will be available when we expect to check for the explain flag.